org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/InstanceOfExpression.java

```diff
@@ -86,13 +86,6 @@ public FlowInfo analyseCode(BlockScope currentScope, FlowContext flowContext, Fl
 	if (initsWhenTrue == null) {
 		flowInfo = this.expression.analyseCode(currentScope, flowContext, flowInfo).
 				unconditionalInits();
-		if (this.elementVariable != null) {
-			initsWhenTrue = flowInfo.copy();
-		}
-	}
-	if (this.elementVariable != null) {
-		initsWhenTrue.markAsDefinitelyAssigned(this.elementVariable.binding);
-		initsWhenTrue.markAsDefinitelyNonNull(this.elementVariable.binding);
 	}
```

Comment:
This is unnecessary - descending into the pattern node achieves the same and is the right way to do this.